packages/core/src/runtime/resolveResetStyleRules.ts

```diff
@@ -0,0 +1,122 @@
+import hashString from '@emotion/hash';
+import { convertProperty } from 'rtl-css-js/core';
+
+import { RESET_HASH_PREFIX } from '../constants';
+import { GriffelStyle, GriffelResetStyle } from '../types';
+import { compileCSSRules, normalizePseudoSelector } from './compileCSS';
+import { isMediaQuerySelector } from './utils/isMediaQuerySelector';
+import { isLayerSelector } from './utils/isLayerSelector';
+import { isNestedSelector } from './utils/isNestedSelector';
+import { isSupportQuerySelector } from './utils/isSupportQuerySelector';
+import { isObject } from './utils/isObject';
+import { hyphenateProperty } from './utils/hyphenateProperty';
+
+/**
+ * @internal
+ */
+export function createStringFromStyles(styles: GriffelResetStyle) {
+  let ltrCSS = '';
+  let rtlCSS = '';
+
+  // eslint-disable-next-line guard-for-in
```

Comment:
is this for performance reasons? the `styles` here is provided by the user, this might leak some weird things if someone is tweaking around a custom object as styles.

Reply:
> is this for performance reasons?

Yes, we use it consistently across in other places, too. It's not expected to see custom objects there, but we can change it if someone will report issues.